api/graphql-client.go

```
@@ -161,7 +162,11 @@ func (thisRef GraphQLClient) FindDeviceByUID(uid string) (Device, error) {
 		}
 	}
 
-	return devices[0], nil
+	if len(devices) > 0 {
+		return devices[0], nil
+	}
+
+	return Device{}, errors.New("device not found")
```

Do we want to return an error here? How is this error used?

---

It is used by the github.com/remoteit/cli/connections/localproxy/through-remoteit-proxy.go
in `graphQLClient.FindDeviceByUID(initiator.UID)`